data/repository/src/main/java/com/cupcake/repository/BaseRepository.kt

```diff
@@ -20,8 +21,8 @@ abstract class BaseRepository {
                 throw ErrorType.Server(baseResponse?.message!!)
             }
         } else {
-            val errorResponse = response.errorBody()?.toString()
-            throw ErrorType.Server(errorResponse ?: "Error Network")
+            val errorResponse = JSONObject(response.errorBody()?.string()!!)
```

Comment:
```suggestion
            val errorResponse = JSONObject(response.errorBody()?.string().toString())
```
this is better and safe I guess